contracts/strategy/LinearYieldDistributionStrategy.sol

```diff
@@ -0,0 +1,173 @@
+// SPDX-License-Identifier: MIT
+pragma solidity =0.8.10;
+
+import {IERC20} from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+
+import {IStrategy} from "./IStrategy.sol";
+import {BaseStrategy} from "./BaseStrategy.sol";
+
+abstract contract LinearYieldDistributionStrategy is BaseStrategy {
+    /**
+     * Emmited when a new yield distribution cycle starts.
+     *
+     * @param startTimestamp the timestamp of the start of the current yield distribution cycle.
+     * @param duration the duration of the current yield distribution cycle
+     * @param distributionAmount the amount being distributed in the current cycle.
+     * @param startAmount the sum of deposits and distributed yield at the start of the cycle.
+     */
+    event StrategyNewYieldDistributionCycle(
+        uint256 startTimestamp,
+        uint256 duration,
+        uint256 distributionAmount,
+        uint256 startAmount
+    );
+
+    error StrategyInvalidYieldDistributionCycleDuration();
+
+    uint256 public constant MIN_CYCLE_DURATION = 1 days;
+    uint256 public constant MAX_CYCLE_DURATION = 21 days;
```

Comment:
do we want to make these configurable?

Reply:
as we discussed, made [changes](https://github.com/lindy-labs/sc_solidity-contracts/pull/241/commits/2febfc16e2ea210a12a46db769d753842cc37922) to allow for 0 min duration to effectively disable yield distribution